### ISSUE TITLE
Water on ship causing crash

### ISSUE THREAD

**Issue by Sunspot0411** [ValkyrienSkies/Valkyrien-Skies#337]
I find out when there's water on a ship, if you look straight up on that ship, the server will crash

**Comment by DeltaNedas** [comment]
thats good and all but
**VERSION**
**LOGS** (.minecraft/logs/latest.log)

**Comment by Sunspot0411** [comment]
I use build 462, the curse 0.9.2 release version. 
[latest.txt](https://github.com/ValkyrienSkies/Valkyrien-Skies/files/3650829/latest.txt)

**Comment by DeltaNedas** [comment]
outdated and unsupported probably, use latest circleci build

**Comment by Sunspot0411** [comment]
no, 462 is not quite outdated, it is a pretty recent release for the master branch. circleci releases are breaking/1.0 branch releases, which is unstable itself, not to mention it is not compatible with plasma engines. I AM sure this problem is not fixed yet.

**Comment by DeltaNedas** [comment]
[02:23:34] [Client thread/WARN] [ic2.EnergyNet]: EnergyTileLoadEvent: posted for com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@6944af45 (dim 0: 5120297/121/743) client-side, aborting
[02:23:34] [Client thread/WARN] [ic2.EnergyNet]: EnergyTileLoadEvent: posted for com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@92fa01a (dim 0: 5120297/121/761) client-side, aborting
[02:23:34] [ic2-poolthread-2/WARN] [ic2.EnergyNet]: Tile com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@71a075dc (dim 0: 5120297/121/743) is already registered
[02:23:34] [ic2-poolthread-2/WARN] [ic2.EnergyNet]: Tile com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@52e9c7d4 (dim 0: 5120297/121/761) is already registered
[02:23:35] [ic2-poolthread-8/WARN] [ic2.EnergyNet]: Tile com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@71a075dc (dim 0: 5120297/121/743) is already registered
[02:23:35] [ic2-poolthread-8/WARN] [ic2.EnergyNet]: Tile com.ice2670.plasmacannon.tileentities.TileEntityPlasmaCannon@52e9c7d4 (dim 0: 5120297/121/761) is already registered
[02:23:35] [Server thread/INFO] [net.minecraft.network.NetHandlerPlayServer]: Keystone2670 lost connection: ???????????????
[02:23:35] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Keystone2670?????

like i said, use latest circleci
this is a plasma engines issue

**Comment by Rubydesic** [comment]
Can you replicate this without plasma engines?

**Comment by Rubydesic** [comment]
And are you sure this is the logs from the crash? 

```
[02:23:35] [Server thread/INFO] [net.minecraft.network.NetHandlerPlayServer]: Stopping singleplayer server as player logged out
[02:23:35] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Stopping server
[02:23:35] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving players
[02:23:35] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving worlds
[02:23:35] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving chunks for level 'world'/overworld
```

It indicates that your game was stopped by logging out - that might just mean its some weird kind of crash, but I just want to make sure.

**Comment by DeltaNedas** [comment]
i believe thats a serverside only crash and will kick you to server list

**Comment by Sunspot0411** [comment]
It has to be log out because I forced the game to shut down. The game was frozen and did not respond before I use task manager to force it stop

**Comment by Sunspot0411** [comment]
@DeltaNedas it is a client crash. The player gets kicked out when look straight up.  but if there's no water on ship this would not happen.

**Comment by Sunspot0411** [comment]
Similar problem also exist if there's a activated ic2 nulcear reactor on ship

**Comment by Sunspot0411** [comment]
here's the log with no plasma cannon on ship
[latest.txt](https://github.com/ValkyrienSkies/Valkyrien-Skies/files/3694818/latest.txt)
[debug.txt](https://github.com/ValkyrienSkies/Valkyrien-Skies/files/3694819/debug.txt)
i'm not sure what was causing this. client-side will kick player out if he/she looks straight up. I tried a small ship with both water and a charging nuclear reactor on the ship it turned out to be fine. however, it still happens with a large ship.

**Comment by DeltaNedas** [comment]
please try again with **JUST** valkyrien skies and forge, maybe optifine too.
you still have plasma cannons installed

**Comment by DeltaNedas** [comment]
also just saying torocraft is spamming up your log, please minimally reproduce it

**Comment by Sunspot0411** [comment]
just vs don't make sense, because no one is gonna install purely vs mod

**Comment by Sunspot0411** [comment]
@DeltaNedas please, my ship is made of plasma cannon blocks, deleting plasma cannons would destroy the ship. And why you think plasma cannon mod is causing this issue? I am sure plasma cannon is not the problem, cause i made that mod. if you don't have the ability to fix or find the cause of the issue then stop saying useless words

**Comment by DeltaNedas** [comment]
1. it does make sense, if vs is not at fault then it is not our problem.
2. try it out on a new world, with just vs, forge and optifine.

**Comment by Sunspot0411** [comment]
no, to do so you exclude compatibility fixing, which is not OK in general. Not to mention plasma cannon is a mod designed for vs, it's a module of vs. so it has to be there

**Comment by DeltaNedas** [comment]
then fix it yourself, you made it

**Comment by Sunspot0411** [comment]
@DeltaNedas and as i told you, I'm sure plasma cannon is not the source of this problem

**Comment by DeltaNedas** [comment]
then recreate it without it jfc

**Comment by Sunspot0411** [comment]
@DeltaNedas why do you keep thinking plasma cannon is the problem? do you even know how to locate problems by reading log?

**Comment by Sunspot0411** [comment]
That's what i did for the new log, deleting plasma cannon mod is not possible

**Comment by DeltaNedas** [comment]
1. remove torocraft, it keeps spamming up a 10(!)MB log
2. use only vs forge and optifine
3. make a ship with water in a new world
4. see if issue occurs
if you can locate problems by reading your logs, please do

**Comment by Sunspot0411** [comment]
only vs forge and optifine don't make sense at all, because as I said, to do so you are ignoring compatibility completely!

**Comment by DeltaNedas** [comment]
it is to verify that the issue is vs itself since you are hellbent on plasma not being the cause

**Comment by Sunspot0411** [comment]
@DeltaNedas you are not the core dev of vs,  so your words really don't count. I post log here is for core dev aka thebest108 to take a look and try find out what went wrong. Even if the problem is probably not caused by vs, i still need vs dev to figure it out, because I don't know how vs is coded

**Comment by Sunspot0411** [comment]
don't you see? it can also be a compatibility issue! if that's the case then testing with only vs, forge and optifine won't recreate the problem and won't help at all. do you expect ppl to play with vs as the only mod? if not then stop asking ppl to test bugs with only vs, forge and optifine

**Comment by DeltaNedas** [comment]
1. he isnt the only dev, see ruby and pork
2. ive contributed already
3. if you want to know how vs works and fix your mod, see https://github.com/ValkyrienSkies/Valkyrien-Skies/
please, for the love of all that is needful, VERIFY THAT THE ISSUE IS CAUSED BY VALKYRIEN SKIES AND NOT YOUR MOD, IF IT IS YOUR MOD YOU OBVIOUSLY NEED TO FIX IT YOURSELF JFC
playing with a minimal setup ensures that the issue is easy to replicate and to pinpoint exactly what causes it
if it is a compatibility issue, then you will not find out with a spammed up log which was killed, you will need to test mods one at a time

**Comment by Sunspot0411** [comment]
ruby does know a bit, pork almost knows nothing about how vs work. I bet you also don't how vs codes work. contributing to the mod doesn't mean you understand the mod

**Comment by DeltaNedas** [comment]
can you please verify that valkyrien skies is the root of the issue?

**Comment by Sunspot0411** [comment]
no, I have 39mods installed. but even if vs is not the root, i still need vs dev to figure out which part of vs had the conflict. because this problem only happens on ship. players not on ship doesn't have this issue

**Comment by Sunspot0411** [comment]
and not only water, charging ic2 nulear generator cause the same issue, and this issue seems only happen on large ship

**Comment by DeltaNedas** [comment]
THEN FIND OUT WHAT ACTUALLY CAUSED THE ISSUE WE CANT WE DO NOT HAVE ACCESS TO YOUR SETUP

**Comment by Sunspot0411** [comment]
That can't be down without vs core dev! cause I don't know how vs codes work and this only happen on ship! it's definitely some part of vs is screwed! either caused by vs itself or incompatibility with other mods or simply because there are too many blocks on ship! you don't have they ability to do so then I'm gonna ask other devs, like Ruby

**Comment by DeltaNedas** [comment]
ok smarty
i tested it out
it is **NOT** caused by vs.
see pic related 
![proof](https://user-images.githubusercontent.com/39013340/66608865-71a5b480-ebaf-11e9-8bfc-78469d03ef2b.png)
fix your own mod
rubert please close

**Comment by Sunspot0411** [comment]
Wrong way of inspection, as I said you are ignoring compatibility if you have only forge, vs and optifine. I did inspection, this error is not caused by water or any specific block. For that I should change the title. On a small ship there would be no error even if plasma cannons,  ic2 machines, omt turrets are all on board. Only when ship is large like about 40000 blocks large would have such a problem

**Comment by Sunspot0411** [comment]
@DeltaNedas I'm not asking you to inspect this problem, you don't have enough knowledge of vs to fix this. so stop paying attention to this issue, it's none of your business. I'm asking you as a member of the devs

**Comment by Sunspot0411** [comment]
@DeltaNedas by the way, if the game has only forge, plasam cannon, ic2, optifine and any other 30 plus mods, such a crash does not happen. so it's definitely related to vs one way or another. most likely it's some compatibility problem

**Comment by DeltaNedas** [comment]
pleae upload a save, you didn't mention a ship being that large

**Comment by Sunspot0411** [comment]
that save needs IC2, plasma cannons, plasma engines, open modular turrets and vs all installed

**Comment by DeltaNedas** [comment]
what save

**Comment by Sunspot0411** [comment]
the save with my 40000 blocks ship

**Comment by DeltaNedas** [comment]
where

**Comment by Sunspot0411** [comment]
I haven't upload, if you can do the test than I'll post it on google drive with all the needed mods

**Comment by DeltaNedas** [comment]
please do that

**Comment by Sunspot0411** [comment]
OK I'll do that when back home

**Comment by Sunspot0411** [comment]
https://drive.google.com/open?id=13wG1mOTvzZK4f6qfNQU1Pj0yEnV8CqT4 
I use a relatively old version of vw so that I can keep all ic2 blocks. I don't think this issue is fixed for newest version.  remember to try look straight up(look up tp MAX angle) many times, this problem is not that easy to reproduce
